nibabel/externals/netcdf.py

```diff
@@ -607,7 +607,10 @@ def _read_var_array(self):
                     pos = self.fp.tell()
                     self.fp.seek(begin_)
                     data = fromstring(self.fp.read(a_size), dtype=dtype_)
-                    data.shape = shape
+                    # Avoid exception for empty variables.
```

Comment:
This is a generic issue with this NetCDF library, in that a NetCDF variable can have a data offset pointing to the end of the file. I've created a test case, but the current code fails if mmap is enabled. I have a more elaborate fix that avoids these failures.